File: 0014-romanToInteger/13-RomanToInteger.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mp={
            "I":1,
            "V":5,
            "X":10,
            "L":50,
            "C":100,
            "D":500,
            "M":1000
        }


        i=0
        num=0

        while i<len(s):
            if i+1<len(s) and mp[s[i+1]]>mp[s[i]]:
                num+=mp[s[i+1]]-mp[s[i]]
                i+=2
            else:
                num+=mp[s[i]]
                i+=1
            
        return num


        

       

File: 0013-RomanToInteger/13-RomanToInteger.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> myMap= new HashMap<>();
        int sum = 0 ; 
        int prev= 0 ; 
        myMap.put('I', 1);
        myMap.put('V', 5);
        myMap.put('X', 10);
        myMap.put('L', 50);
        myMap.put('C',100);
        myMap.put('D',500);
        myMap.put('M',1000);

        for(char c : s.toCharArray()){
            sum += myMap.get(c);
            if(prev < myMap.get(c)){
                sum = sum - 2 * prev; 
            }
            prev = myMap.get(c);
            
        }
        return sum ;


        
    }
}